geode-core/src/main/java/org/apache/geode/cache/query/security/JavaBeanAccessorMethodAuthorizer.java

```diff
@@ -0,0 +1,173 @@
+  static final String NULL_CACHE_MESSAGE = "Cache should be provided to configure the authorizer.";
+  static final String GEODE_BASE_PACKAGE = "org.apache.geode";
+  private final RestrictedMethodAuthorizer restrictedMethodAuthorizer;
+  private final Set<String> allowedPackages;
+
+  /**
+   * Creates a {@code JavaBeanAccessorMethodAuthorizer} object and initializes it so it can be
+   * safely used in a multi-threaded environment.
+   * <p/>
+   *
+   * Applications can use this constructor as part of the initialization for custom authorizers
+   * (see {@link Declarable#initialize(Cache, Properties)}), when using a declarative approach.
+   *
+   * @param cache the {@code Cache} instance that owns this authorizer, required in order to
+   *        configure the default {@link RestrictedMethodAuthorizer}.
+   * @param allowedPackages the packages containing classes for which 'is' and 'get' methods will
+   *        be authorized.
+   */
+  public JavaBeanAccessorMethodAuthorizer(Cache cache, Set<String> allowedPackages) {
+    Objects.requireNonNull(cache, NULL_CACHE_MESSAGE);
+    Objects.requireNonNull(allowedPackages, NULL_PACKAGE_MESSAGE);
+    this.restrictedMethodAuthorizer = new RestrictedMethodAuthorizer(cache);
+    this.allowedPackages = Collections.unmodifiableSet(allowedPackages);
+  }
+
+  /**
+   * Creates a {@code JavaBeanAccessorMethodAuthorizer} object and initializes it so it can be
+   * safely used in a multi-threaded environment.
+   * <p/>
+   *
+   * @param restrictedMethodAuthorizer the default {@code RestrictedMethodAuthorizer} to use.
+   * @param allowedPackages the packages containing classes for which 'is' and 'get' methods will
+   *        be authorized.
+   */
+  public JavaBeanAccessorMethodAuthorizer(RestrictedMethodAuthorizer restrictedMethodAuthorizer,
+      Set<String> allowedPackages) {
+    Objects.requireNonNull(restrictedMethodAuthorizer, NULL_AUTHORIZER_MESSAGE);
+    Objects.requireNonNull(allowedPackages, NULL_PACKAGE_MESSAGE);
+    this.restrictedMethodAuthorizer = restrictedMethodAuthorizer;
+    this.allowedPackages = Collections.unmodifiableSet(allowedPackages);
+  }
+
+  /**
+   * Executes the authorization logic to determine whether the {@code method} is allowed to be
+   * executed on the {@code target} object instance.
+   * <p/>
+   *
+   * @param method the {@link Method} that should be authorized.
+   * @param target the {@link Object} on which the {@link Method} will be executed.
+   * @return {@code true} if the {@code method} can be executed on on the {@code target} instance,
+   *         {@code false} otherwise.
+   *
+   * @see org.apache.geode.cache.query.security.MethodInvocationAuthorizer
+   */
+  @Override
+  public boolean authorize(Method method, Object target) {
+
+    // First ensure that the given method exists for the given target
+    if (!Arrays.asList(target.getClass().getMethods()).contains(method)) {
+      return false;
+    }
```

Comment:
Even though is not a bad idea to have this, is not needed and will just incur into a waste time during the query execution. The query engine is responsible for invoking the `MethodInvocationAuthorizer` and, internally, it always gets the actual `method` instance from the actual `target` object through the usage of reflection, so the condition will always evaluate to `true`.